document new options in dbt-sqlserver 1.4.1 (#3405)
fishtown-analytics/docs.getdbt.com: website/docs/reference/warehouse-setups/mssql-setup.md

If you already have ODBC Driver 17 installed, then that one will work as well.

#### Supported configurations

* The adapter is tested with SQL Server 2017, SQL Server 2019, SQL Server 2022 and Azure SQL Database.
* We test all combinations with Microsoft ODBC Driver 17 and Microsoft ODBC Driver 18.
* The collations we run our tests on are `SQL_Latin1_General_CP1_CI_AS` and `SQL_Latin1_General_CP1_CS_AS`.

The adapter support is not limited to the matrix of the above configurations. If you notice an issue with any other configuration, let us know by opening an issue on [GitHub](https://github.com/dbt-msft/dbt-sqlserver).

## Authentication methods & profile configuration

### Common configuration

For all the authentication methods below, the following configuration options can be set in your `profiles.yml` file. The options marked with `*` are required.
A complete reference of all options can be found [at the end of this page](#reference-of-all-connection-options).

* `driver`*: The ODBC driver to use. E.g. `ODBC Driver 18 for SQL Server`
* `server`*: The server hostname. E.g. `localhost`
* `port`: The server port. E.g. `1433`
* `database`*: The database name.

> * `schema`*: The schema name. E.g. `dbo`

Reviewer: ```suggestion
* `schema`*: The schema name. For example, `dbo`
```

Download and install the [Microsoft ODBC Driver 18 for SQL Server](https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver15).
If you already have ODBC Driver 17 installed, then that one will work as well.

#### Supported configurations

* The adapter is tested with SQL Server 2017, SQL Server 2019, SQL Server 2022 and Azure SQL Database.
* We test all combinations with Microsoft ODBC Driver 17 and Microsoft ODBC Driver 18.
* The collations we run our tests on are `SQL_Latin1_General_CP1_CI_AS` and `SQL_Latin1_General_CP1_CS_AS`.

The adapter support is not limited to the matrix of the above configurations. If you notice an issue with any other configuration, let us know by opening an issue on [GitHub](https://github.com/dbt-msft/dbt-sqlserver).

## Authentication methods & profile configuration

### Common configuration

> For all the authentication methods below, the following configuration options can be set in your `profiles.yml` file. The options marked with `*` are required.

Reviewer: ```suggestion
For all the authentication methods, refer to the following configuration options that can be set in your `profiles.yml` file. 
```
